src/System.Private.CoreLib/shared/System/Convert.cs

```diff
@@ -2518,21 +2520,127 @@ public static unsafe bool TryToBase64Chars(ReadOnlySpan<byte> bytes, Span<char>
                 charsWritten = ConvertToBase64Array(outChars, inData, 0, bytes.Length, insertLineBreaks);
                 return true;
             }
+        }      
+
+        internal static readonly Vector128<byte> s_base64ShuffleMask = Vector128.Create((byte)
+            1, 0, 2, 1, 4, 3, 5, 4, 7, 6, 8, 7, 10, 9, 11, 10);
+
+        internal static readonly Vector128<byte> s_base64ShiftLut = Vector128.Create(
+            (sbyte)'a' - 26, (sbyte)'0' - 52,
+            (sbyte)'0' - 52, (sbyte)'0' - 52,
+            (sbyte)'0' - 52, (sbyte)'0' - 52,
+            (sbyte)'0' - 52, (sbyte)'0' - 52,
+            (sbyte)'0' - 52, (sbyte)'0' - 52,
+            (sbyte)'0' - 52, (sbyte)'+' - 62,
+            (sbyte)'/' - 63, (sbyte)'A', 0, 0).AsByte();
+
+        internal static readonly Vector128<byte> s_base64TwoBytesStringMaskLo = Vector128.Create(
+                0, 0x80, 1, 0x80,
+                2, 0x80, 3, 0x80,
+                4, 0x80, 5, 0x80,
+                6, 0x80, 7, 0x80);
+
+        private static unsafe (int i, int j, int charcount) ConvertToBase64ArraySsse3(char* outChars, byte* inData, int length, int offset, bool insertLineBreaks)
```

Comment:
Returning a value tuple for larger inputs (`length >>`) maybe no (perf-) problem, but for small input lengths it could be better to use `out`, so that the JIT may use registers instead of the stack. Though here I'm not sure what the JIT emits, as there a quite a lot of arguments involved.

Reply:
Example: https://gist.github.com/EgorBo/4f62e9da70f2eb6b61cc3334da709764
Asm diff: 
![image](https://user-images.githubusercontent.com/523221/52631922-2973ac80-2ed1-11e9-91af-abcd96b0470f.png)